#! /usr/bin/env python

# Copyright (c) 2011, Markus Achtelik, ASL, ETH Zurich, Switzerland
# You can contact the author at <markus dot achtelik at mavt dot ethz dot ch>
# 
# All rights reserved.
# 
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions are met:
# * Redistributions of source code must retain the above copyright
# notice, this list of conditions and the following disclaimer.
# * Redistributions in binary form must reproduce the above copyright
# notice, this list of conditions and the following disclaimer in the
# documentation and/or other materials provided with the distribution.
# * Neither the name of ETHZ-ASL nor the
# names of its contributors may be used to endorse or promote products
# derived from this software without specific prior written permission.
# 
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
# DISCLAIMED. IN NO EVENT SHALL ETHZ-ASL BE LIABLE FOR ANY
# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.


PACKAGE='asctec_hl_interface'
import roslib; roslib.load_manifest(PACKAGE)

#from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

# @todo Think about levels. Setting most of these to STOP to guarantee atomicity.
# @todo Double check these ranges, defaults

MISC                = gen.const("MISC",             int_t, 0x01000000, "misc")

#       Name                        Type        Reconfiguration level    Description                                           Default Min     Max
gen.add("cancel_wp",                 bool_t,     MISC["value"],     "aborts the current waypoint and waypoint list",             False)
gen.add("pos_var",                 double_t,   MISC["value"],     "waypoint position accuracy to mark as reached [m]",         0.5,	0,		5)
gen.add("yaw_var",                 double_t,   MISC["value"],     "waypoint yaw accuracy to mark as reached [deg]",         	 1,		0,		45)
gen.add("goal_yaw",                 double_t,   MISC["value"],     "constant goal point yaw",         	 						 0,		0,		360)
gen.add("timeout",                 double_t,   MISC["value"],     "timeout to reach waypoint [s]",          				     10,		0,		60)


exit(gen.generate(PACKAGE, "Config", "WaypointNode"))
